src/Pos/Block/Network/Listeners.hs

```diff
@@ -78,24 +79,23 @@ stubListenerConv' = unproxy $ \(_ :: Proxy ssc) ->
 handleGetHeaders
     :: forall ssc m.
        (WorkMode ssc m)
-    => (ListenerSpec m, OutSpecs)
-handleGetHeaders = listenerConv $ \_ peerId conv -> do
-    logDebug $ "handleGetHeaders: request from " <> show peerId
-    handleHeadersCommunication conv
+    => m (ListenerSpec m, OutSpecs)
+handleGetHeaders = reifyMsgLimit (Proxy @MsgGetHeaders) $ \limitProxy ->
```

Comment:
It won't work, I have to first get value, then `reify` it in type, then that type would become available.
From compiler's point of view, `reify` (and `reifyMsgLimit`) gives me `forall s. Reifies s limit => s` type, and `Reifies` constraint is needed to `reflect` value from type back; so `s` is defined under `reifyMsgLimit` only